sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -877,6 +881,34 @@ public CosmosClientBuilder sessionRetryOptions(SessionRetryOptions sessionRetryO
         return this;
     }
 
+    /**
+     * Sets a {@link Supplier<CosmosExcludedRegions>} which returns a {@link CosmosExcludedRegions} instance when {@link Supplier#get()} is invoked.
+     * The request will not be routed to regions present in {@link CosmosExcludedRegions#getExcludedRegions()}
+     * for hedging scenarios and retry scenarios for the workload executed through this instance
+     * of {@link CosmosClient} / {@link CosmosAsyncClient}.
+     *
+     * @param excludedRegionsSupplier the supplier which returns a {@code CosmosExcludedRegions} instance.
+     * @return current CosmosClientBuilder.
+     * */
+    public CosmosClientBuilder excludedRegionsSupplier(Supplier<CosmosExcludedRegions> excludedRegionsSupplier) {
+        this.cosmosExcludedRegionsSupplier = excludedRegionsSupplier;
+        return this;
+    }
+
+    /**
```

Comment:
I don't quite understand why we need the getter returning the set - why not just have a getter for the supplier (if we even need the getter). But to be conistent with other properties having a getter would be good

Reply:
Ah - never mind it is internal.